onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -464,14 +556,15 @@ std::vector<std::unique_ptr<ComputeCapability>> OpenVINOExecutionProvider::GetCa
 #endif
 
   int counter = 0;
+  std::string error_msg = "";
   std::unique_ptr<IndexedSubGraph> sub_graph = std::make_unique<IndexedSubGraph>();
 
   auto model_proto = GetModelProtoFromFusedNode(graph_viewer);
 
   std::set<const onnxruntime::NodeArg*> fused_inputs, fused_outputs;
 
-  if (!IsGraphSupported(graph_viewer,device_id)) {
-    LOGS_DEFAULT(WARNING) << openvino_ep::OpenVINOGraph::log_tag << "Rejecting as graph has unsupported operations.";
+  if (!IsGraphSupported(graph_viewer,device_id, error_msg)) {
```

Comment:
With the exception semantics, the usage here would be:
try {
      CheckGraphSupport(graph_viewer, device_id);
} catch (const char* msg) {
   LOGS_DEFAULT(WARNING) << openvino_ep::OpenVINOGraph::log_tag << "Rejecting graph: " << msg;
}